simple-streaming-app/src/main/scala/com/typesafe/spark/SimpleStreamingApp.scala

```diff
@@ -9,20 +9,39 @@ import scala.concurrent.ExecutionContext.Implicits.global
 import scala.io
 import com.typesafe.spark.test.Hanoi
 import org.apache.spark.storage.StorageLevel
+import org.apache.spark.streaming.dstream.DStream
+import scopt.OptionParser
+import scopt.Zero
+
+/** Simple statistics data class
+ */
+private case class Stats(count: Int, sum: Long, mean: Double, stdDev: Double, millis: Long)
 
 object SimpleStreamingApp {
 
   def main(args: Array[String]): Unit = {
 
-    val (hostname, port, strategy, batchInterval) = parseArgs(args)
+    val config = parseArgs(args)
 
+    {
+      import config._
+      println(s"""
+        Connecting to $hostname:$port using Spark $master and $streams receivers.
+      """)
+    }
     val conf = new SparkConf()
       .setAppName("Streaming tower of Hanoi resolution")
-      .set("spark.streaming.receiver.congestionStrategy", strategy)
+      .setMaster(config.master)
```

Comment:
If it can be used only if it is not already define in the Spark config, I think it would be good. So it can be run simply as an app, and also as a job.